### PR TITLE
LGA-A3459 - Add user creation/update/removal management commands

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Add user creation/update/removal management commands.

## Any other changes that would benefit highlighting?

Added the `tabulate` package to print tables when listing models in management commands

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by TawneeOwl — APPROVED** [review]
This is awesome!!!

File: docs/source/documentation/user-management.erb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: User Management
+---
+
+## Adding users
```
Comment (digitalali-moj) [inline docs/source/documentation/user-management.erb:5]: Is this to be run from the environment pod container?

File: manage.py [hunk 2]
```diff
@@ -1,57 +1,149 @@
...
 
-session: Session = next(get_session())
+
+def init_session(typer_app: typer.Typer, session: Session) -> None:
+    typer_app.db_session = session
 
 
 @app.command()
-def user_scopes_add(
+def add_scopes(
```
Comment (digitalali-moj) [inline manage.py:26]: Should this be add set_user_scopes so we don't get confused with scope diagnosis also it's replacing the whole scope?

File: manage.py [hunk 3]
```diff
@@ -1,57 +1,149 @@
...
+
 app = typer.Typer()
 
-session: Session = next(get_session())
+
+def init_session(typer_app: typer.Typer, session: Session) -> None:
+    typer_app.db_session = session
 
 
 @app.command()
```
Comment (digitalali-moj) [inline manage.py:25]: Do we need documentation and tests around scopes or is that a separate thing / done already?

File: manage.py [hunk 4]
```diff
@@ -1,57 +1,149 @@
...
+def add_scopes(
     username: str, scope: Annotated[List[UserScopes], typer.Option()]
 ) -> None:
     statement = select(User).where(User.username == username)
-    user: User = session.exec(statement).first()
+    user: User = app.db_session.exec(statement).first()
     print(
         f"Replacing user {user.username} current scopes {user.scopes} with new scopes {scope}..."
     )
     user.scopes = scope
```
Comment (digitalali-moj) [inline manage.py:34]: Do we need confirmation message here and for the standard update-user as either could invalidate access to the API.

**Review by digitalali-moj — APPROVED** [review]
Great work thanks - few questions to consider

File: manage.py (hunk 2 above)
Comment (said-moj) [reply manage.py:26]: Good idea, I will change it to `add_user_scopes` to keep it consistent(add_x/list_x/update_x)

File: manage.py (hunk 3 above)
Comment (said-moj) [reply manage.py:25]: I think we already have scopes docs `https://github.com/ministryofjustice/laa-civil-case-api/blob/main/docs/source/documentation/scopes.erb`

Comment (said-moj) [reply manage.py:25]: Added tests for user scopes now

File: manage.py (hunk 4 above)
Comment (said-moj) [reply manage.py:41]: Added confirmation message to update operation

**Review by TawneeOwl — APPROVED** [review]
LGTM

File: docs/source/documentation/user-management.erb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+Use `./manage.py add-user` to add users
+
+```
+ Usage: manage.py add-user [OPTIONS] USERNAME
+
+╭─ Arguments ──────────────────────────────────────────────────────────────────────────────────────────────────╮
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:11]: This file doesn't seem to get correctly rendered when viewed on the web.

This is what it looks like when I build it locally:

Screenshot 2024-12-18 at 13 42 51

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]

File: docs/source/documentation/user-management.erb [hunk 6]
```diff
@@ -0,0 +1,58 @@
+---
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:1]: This file needs to be named `user-management.html.md.erb` in order for it to be rendered as expected.

File: docs/source/documentation/user-management.erb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+*  --full-name                    TEXT  [default: None] [required]
+*  --password                     TEXT  [default: None] [required]
+  --disable      --no-disable          [default: no-disable]
+  --help                               Show this message and exit.
+
+
+Example: ./manage.py add-user test_user --email test.user@digital.justice.gov.uk --full-name "Test User" --password "password123"
+```
+
+##Updating users
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:25]: There needs to be a space after these hashtags for this heading to be rendered correctly.

File: docs/source/documentation/user-management.erb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+Use `./manage.py update-user` to update users
+```
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:27]: There needs to be a linebreak before the code block for this to be rendered as expected.

File: docs/source/documentation/user-management.erb [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+ --password                     TEXT  [default: None]
+ --disable      --no-disable          [default: no-disable]
+ --enable       --no-enable           [default: no-enable]
+ --help                               Show this message and exit.
+
+
+Example: ./manage.py update--user test_user --email test.user@justice.gov.uk --full-name "Sir Test User"
+```
+
+##Delete users
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:45]: Likewise here

File: docs/source/documentation/user-management.erb [hunk 10]
```diff
@@ -0,0 +1,58 @@
+---
+title: User Management
+---
+
```
Comment (BenMillar-MOJ) [inline docs/source/documentation/user-management.erb:4]: There should be a H1 here to populate the rendered page title:
`# User Management`

**Review by BenMillar-MOJ — CHANGES_REQUESTED** [review]

**Review by BenMillar-MOJ — APPROVED** [review]
LGTM

**Pull request merged**